ground/src/main/java/com/google/android/ground/model/mutation/MediaMutation.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.model.mutation
+
+import java.util.Date
+
+/**
+ * Represents a local change to media associated with a submission.
+ *
+ * These mutations are directly dependent upon another mutation type: SubmissionMutation. At
+ * present, any MediaMutation will only have relevance in the context of a submission. To this end,
+ * several properties are derived from the mutation's associated submission mutation.
+ */
+data class MediaMutation(
+  override val type: Type,
+  val mediaId: String,
+  val submissionMutation: SubmissionMutation,
+  override val id: Long? = null,
+  override val syncStatus: SyncStatus = SyncStatus.PENDING,
+  override val retryCount: Long = 0,
+  override val lastError: String = "",
+) : Mutation() {
```

Comment:
Not sure where this should be represented so dropping here for now: at present there's only one type of MediaMutation: "upload" mutation (create?), but no delete update.

Reply:
Good point! Maybe we want a separate enum here. This might be more evidence that this type is not *really* a complete mutation type, and maybe we should just move it out of the subtype tree. though certain properties (rerties, sync status, id) are certainly shared.

Reply:
It may or may not be more elegant to model all remote changes as mutations - in this case the mutation is the CREATE a new media entity in data storage. If we were to all "edit" we could also have an "UPDATE" operation which would just be implemented as a DELETE and CREATE. I'd say let's go with the enum we have for now and keep this in the back of our minds in the next PRS?

Reply:
sgtm